src/components/Home.jsx

```diff
@@ -1,62 +1,108 @@
-                   <p className="filter-by ml-0">Filter-by</p>
-                   <button type="submit" className="btn btn-default tag-button">#Learntech</button>
-                   <div className="tweets">
-                   <img src={Avatar} alt="Avatar" className="avatar" />
-                   <span className="tweet-body">
-                   <span className="tweet-time">2 hours ago</span>
-                    <h6 className="tweet-user">Maya Neria</h6>   
-                    <p class="main-tweet">I love #regle so much - everyone should use it! #learntech</p>
-                   </span>
-                   </div>
-                   <div className="tweets">
-                   <img src={Avatar} alt="Avatar" className="avatar" />
-                   <span className="tweet-body">
-                   <span className="tweet-time">2 hours ago</span>
-                    <h6 className="tweet-user">Maya Neria</h6>   
-                    <p class="main-tweet">I love #regle so much - everyone should use it! #learntech</p>
-                   </span>
-                   </div>
-                   <div className="tweets">
-                   <img src={Avatar} alt="Avatar" className="avatar" />
-                   <span className="tweet-body">
-                   <span className="tweet-time">2 hours ago</span>
-                    <h6 className="tweet-user">Maya Neria</h6>   
-                    <p class="main-tweet">I love #regle so much - everyone should use it! #learntech</p>
-                   </span>
-                   </div>
-                   <div className="tweets">
-                   <img src={Avatar} alt="Avatar" className="avatar" />
-                   <span className="tweet-body">
-                   <span className="tweet-time">2 hours ago</span>
-                    <h6 className="tweet-user">Maya Neria</h6>   
-                    <p class="main-tweet">I love #regle so much - everyone should use it! #learntech</p>
-                   </span>
-                   </div>
+class Index extends Component {
+    constructor (props){
+        super(props)
+        this.state = {
+            value: 280,
+            disabled: false,
+            text_value: ''
+        }
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(event){
+        this.setState({value: 280-event.target.value.length, text_value: event.target.value})
+        if (this.state.value <= 0){
+            this.setState({disabled: true})
+        }else{
+            this.setState({disabled: false})
+        }
+
+    }
+
+    handleSubmit = e => {
+        const { dispatch, history } = this.props;
+        if (!e.target.checkValidity()) {
+            e.preventDefault();
+            e.stopPropagation();
+        }
+        e.preventDefault();
+        const data = {body: this.state.text_value};
+        dispatch(tweetActions.tweet(data, history))
```

Comment:
I would suggest using `mapDispatchToProps` instead of using dispatch fn directly. More reasons here:
https://react-redux.js.org/using-react-redux/connect-mapdispatch#providing-a-mapdispatchtoprops-parameter

Reply:
This has been rectified as well